src/Features/Core/Portable/EditAndContinue/EditAndContinueMethodDebugInfoReader.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            private readonly ISymUnmanagedReader5 _symReader;
+            private readonly int _version;
+
+            public Native(ISymUnmanagedReader5 symReader, int version)
+            {
+                Debug.Assert(symReader != null);
+                Debug.Assert(version >= 1);
+
+                _symReader = symReader;
+                _version = version;
+            }
+
+            public override bool IsPortable => false;
+
+            public override StandaloneSignatureHandle GetLocalSignature(MethodDefinitionHandle methodHandle)
+            {
+                var symMethod = (ISymUnmanagedMethod2)_symReader.GetMethodByVersion(MetadataTokens.GetToken(methodHandle), _version);
+
+                // Compiler generated methods (e.g. async kick-off methods) might not have debug information.
+                return symMethod == null ? default : MetadataTokens.StandaloneSignatureHandle(symMethod.GetLocalSignatureToken());
+            }
+
+            public override EditAndContinueMethodDebugInformation GetDebugInfo(MethodDefinitionHandle methodHandle)
+            {
+                int methodToken = MetadataTokens.GetToken(methodHandle);
+
+                byte[] debugInfo;
+                try
+                {
+                    debugInfo = _symReader.GetCustomDebugInfo(methodToken, _version);
+                }
+                catch (ArgumentOutOfRangeException)
+                {
+                    // Sometimes the debugger returns the HRESULT for ArgumentOutOfRangeException, rather than E_FAIL,
+                    // for methods without custom debug info (https://github.com/dotnet/roslyn/issues/4138).
+                    debugInfo = null;
+                }
+                catch (Exception e) when (FatalError.ReportWithoutCrash(e)) // likely a bug in the compiler/debugger
+                {
+                    throw new InvalidDataException(e.Message, e);
+                }
+
+                try
+                {
+                    ImmutableArray<byte> localSlots, lambdaMap;
+                    if (debugInfo != null)
+                    {
+                        localSlots = CustomDebugInfoReader.TryGetCustomDebugInfoRecord(debugInfo, CustomDebugInfoKind.EditAndContinueLocalSlotMap);
+                        lambdaMap = CustomDebugInfoReader.TryGetCustomDebugInfoRecord(debugInfo, CustomDebugInfoKind.EditAndContinueLambdaMap);
+                    }
+                    else
+                    {
+                        localSlots = lambdaMap = default;
+                    }
+
+                    return EditAndContinueMethodDebugInformation.Create(localSlots, lambdaMap);
+                }
+                catch (InvalidOperationException e) when (FatalError.ReportWithoutCrash(e)) // likely a bug in the compiler/debugger
+                {
+                    // TODO: CustomDebugInfoReader should throw InvalidDataException
```

Comment:
>TODO [](start = 23, length = 4)

Is this TODO for the current PR? or should we track it with a GitHub issue?

Reply:
This was there before. I don't think it's worth filing an issue at this point.